src/core/mac/sub_mac.cpp

```diff
@@ -565,6 +644,59 @@ void SubMac::SetState(State aState)
     }
 }
 
+void SubMac::SetMacKey(uint8_t aKeyIdMode, const uint8_t *aPrevKey, const uint8_t *aCurrKey, const uint8_t *aNextKey)
+{
+    switch (aKeyIdMode)
+    {
+    case Frame::kKeyIdMode0:
+    case Frame::kKeyIdMode2:
+        break;
+    case Frame::kKeyIdMode1:
+        OT_ASSERT(aPrevKey != NULL && aCurrKey != NULL && aNextKey != NULL);
+
+        memcpy(mPrevKey, aPrevKey, kMacKeySize);
+        memcpy(mCurrKey, aCurrKey, kMacKeySize);
+        memcpy(mNextKey, aNextKey, kMacKeySize);
+
+        break;
+
+    default:
+        OT_ASSERT(false);
+        break;
+    }
+
+    VerifyOrExit(!ShouldHandleTransmitSecurity(), OT_NOOP);
```

Comment:
Just a question, not a suggestion:
If we run a RCP instance and try to set Mac Key, this method would be called twice: first time on host and this function wouldn't stop at this line, the second time on RCP and this function would stop here, does it work in this way?

Reply:
Yes. It is intended this way. The radio function in line 670 is intended to set MAC key to Radio. In RCP case, on host side, the radio is RCP. So in posix app, the radio_spinel will be the radio layer and this key will be passed to RCP. In RCP side, the radio layer is the real platform, if the platform driver or hardware supports Tx encryption, this key will be passed to it, if not, Submac will handle the Tx security and the key will not be passed down.

For NCP mode, this line is used to decide if submac need to pass the key the real radio driver.

Reply:
Got it, thanks!